packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,355 @@ void main()
+            if (deltaDepth < thickness * 0.1) {
+                hit1 = 1.0;
+                if (deltaDepth < thickness * 0.01) {
+                    break;
+                }
+            }
+            search1 = currSearch;
+            currSearch = ((search1 + search0) / 2.0);
+        } else {
+            search0 = currSearch;
+            currSearch = ((search1 + search0) / 2.0);
+        }
+    }       
+    // end of the second pass
+      
+    // compute how much the reflection is visible
+    if (hit0 == 0.0){
+        #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+            info.visibilityBackup = 1.0;
+        #endif
+
+    } else {
+        info.miss = false;
+
+        if (length(rayDepth - hitCoordVS) < thickness * 0.05 //){
+                   || dot(normalize(dirVS), normalize(texture2D(normalSampler, uv).xyz)) > -0.085 ){
+            info.visibility = 0.0; // avoid displaying reflection when search level is low (auto intersection)
+        } else {
+            vec2 dCoordScreen = smoothstep(vec2(0.2), vec2(0.6), abs(vec2(0.5, 0.5) - uv)); // HermiteInterpolation
+            info.visibility = texture2D(textureSampler, uv).a // alpha value of the reflected scene position 
+                * abs(dot(normalize(hitCoordVS), normalize( texture2D(normalSampler, uv).xyz))) // to avoid artifacts due to missing information when refleted face is almost perpendicular to the camera view direction
+                
+                // reduce back face limits artifacts
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag + increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag - increment * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag + increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+                * clamp(dot(normalize( texture2D(normalSampler, uv).xyz), normalize( texture2D(normalSampler, (currFrag - increment.yx * resolution)/texSize).xyz)), 0.0, 1.0)
+              
+                * (1.0 - max (dot(-normalize(hitCoordVS), normalize(dirVS)), 0.0)) // to fade out the reflection as the reflected direction point to the camera's position (hit behind the camera)
+                * (1.0 - currSearch) // the reflection should be sharper when near from the starting point
+                * (1.0 - clamp (abs(hitCoordVS.z / distanceFade), 0.0, 1.0)) // to fade out the reflection near the distanceFade
+                * (1.0 - clamp (deltaDepth/maxTol, 0.0, 1.0)) // since the hit point is not always precisely found, we fade out the reflected color if we aren't precise enough 
+                * clamp(1.0 - (dCoordScreen.x + dCoordScreen.y), 0.0, 1.0); // to fade out the reflection near the edge of the screen

```

Comment:
I added this factors to deal with this artifacts : https://stackoverflow.com/questions/65760242/fix-artifacts-in-screen-space-reflections
![image](https://user-images.githubusercontent.com/44207625/178489601-6ed22054-9008-4531-ad54-69c57bad8b0d.png)

And, I think the result is prettier with this factors (and using the visibility * 2.0 to compensate)
![image](https://user-images.githubusercontent.com/44207625/178488774-86f93eb5-ca5a-4047-9f58-733fb1057671.png)
![image](https://user-images.githubusercontent.com/44207625/178488788-3d2953af-7b7a-44fb-b96e-c80ae21ace45.png)

Reply:
I wonder in what measure these artifacts come from how the ray marching is done compared to using a robust DDA algorithm like the one described [here](https://jcgt.org/published/0003/04/04/paper.pdf) and used [here](https://github.com/kode80/kode80SSR) and [there](http://casual-effects.blogspot.com/2014/08/screen-space-ray-tracing.html)...

Reply:
Can you share the PG links you used for these screenshots?

I'm doing some testings on my end, hopefully we will be able to come up with something more robust by combining our findings.

Reply:
Hello,
I'm in holyday for a week.
I don't have the playgrounds right now but they looked like that:
https://playground.babylonjs.com/#KA93U#775 
https://playground.babylonjs.com/#PIZ1GK#571

Reply:
Here's what I could come up with using the code [there](https://jcgt.org/published/0003/04/04/paper.pdf):

![image](https://user-images.githubusercontent.com/4152247/180252670-fb3e1a29-40e7-4bb3-8c6c-53a168d73a9e.png)

As you can see, there are a lot less artefacts (only at the base of the ball reflections) and they could be hidden with some blur. I'm going to play a bit more with the code and will make it available in case you are interested in integrating some parts.

![image](https://user-images.githubusercontent.com/4152247/180258177-200f8d7b-e17d-4fd8-80e9-29c122951866.png)